Sources/PackageDescription/Target.swift

```diff
@@ -228,6 +249,22 @@ public final class Target {
                 publicHeadersPath == nil &&
                 pkgConfig == nil &&
                 providers == nil &&
+                extensionCapability == nil &&
+                cSettings == nil &&
+                cxxSettings == nil &&
+                swiftSettings == nil &&
+                linkerSettings == nil
+            )
+        case .extension:
+            precondition(
```

Comment:
these preconditions hint that the model could be improved, maybe something along the lines of what we recently did with PackageDependencyDescription.

Reply:
Agreed.  That should be fixed but seems like a different PR.  The overall approach for this PR is to make the implementation of this new target type be right in line with the existing ones rather than do something different for just this one target type.

Reply:
But yes, this is pretty awful.  We need  a separate effort to rework all of this, for all these targets.